app/main.py

```diff
@@ -1,34 +1,126 @@
+from typing import Any
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.is_drowned = is_drowned
+        self.decks = []
+        for row in range(min(start[0], end[0]),
+                         max(start[0], end[0]) + 1):
+            for column in range(min(start[1], end[1]),
+                                max(start[1], end[1]) + 1):
+                self.decks.append(Deck(row, column))
 
-    def get_deck(self, row, column):
+    def get_deck(self, row: int, column: int) -> None | Deck:
         # Find the corresponding deck in the list
-        pass
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> bool:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                deck.is_alive = False
+                break
+
+        self.is_drowned = all(not deck.is_alive for deck in self.decks)
+        return self.is_drowned
+
+    def is_alive(self, row: int, column: int) -> bool:
```

Comment:
The method name `is_alive` conflicts with the `is_alive` attribute in the `Deck` class. This could lead to confusion or errors. Consider renaming the method to something like `is_deck_alive` to avoid this conflict.